From 742de767290a3ac88a82a6ee6f02fa537fada4c4 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:09 -0400
Subject: [tools] perf/traceevent: Have bprintk output the same as the kernel does

Message-id: <1408462094-14194-261-git-send-email-jolsa@redhat.com>
Patchwork-id: 88217
O-Subject: [PATCH RHEL7.1 BZ1131394 260/325] tools lib traceevent: Have bprintk output the same as the kernel does
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 0883d9d730fc294c3d90ebd190b94e5782ead316
Author: Steven Rostedt (Red Hat) <rostedt@goodmis.org>
Date: Fri Nov 1 17:53:57 2013 -0400

description
===========
The trace_bprintk() in the kernel looks like:

 ring_buffer_producer_thread: Missed:   0
 ring_buffer_producer_thread: Hit:      62174350
 ring_buffer_producer_thread: Entries per millisec: 6296
 ring_buffer_producer_thread: 158 ns per entry
 ring_buffer_producer_thread: Sleeping for 10 secs
 ring_buffer_producer_thread: Starting ring buffer hammer
 ring_buffer_producer_thread: End ring buffer hammer

But the current output looks like this:

 ring_buffer_producer_thread : Time:     9407018 (usecs)
 ring_buffer_producer_thread : Overruns: 43285485
 ring_buffer_producer_thread : Read:     4405365  (by events)
 ring_buffer_producer_thread : Entries:  0
 ring_buffer_producer_thread : Total:    47690850
 ring_buffer_producer_thread : Missed:   0
 ring_buffer_producer_thread : Hit:      47690850

Remove the space between the function and the colon.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-parse.c b/tools/lib/traceevent/event-parse.c
index 0a1ffe0..e1c743c 100644
--- a/tools/lib/traceevent/event-parse.c
+++ b/tools/lib/traceevent/event-parse.c
@@ -3802,8 +3802,8 @@ static struct print_arg *make_bprint_args(char *fmt, void *data, int size, struc
  if (asprintf(&arg->atom.atom, "%lld", ip) < 0)
   goto out_free;
 
- /* skip the first "%pf : " */
- for (ptr = fmt + 6, bptr = data + field->offset;
+ /* skip the first "%pf: " */
+ for (ptr = fmt + 5, bptr = data + field->offset;
       bptr < data + size && *ptr; ptr++) {
   int ls = 0;
 
@@ -3929,12 +3929,12 @@ get_bprint_format(void *data, int size __maybe_unused,
 
  printk = find_printk(pevent, addr);
  if (!printk) {
-  if (asprintf(&format, "%%pf : (NO FORMAT FOUND at %llx)\n", addr) < 0)
+  if (asprintf(&format, "%%pf: (NO FORMAT FOUND at %llx)\n", addr) < 0)
    return NULL;
   return format;
  }
 
- if (asprintf(&format, "%s : %s", "%pf", printk->printk) < 0)
+ if (asprintf(&format, "%s: %s", "%pf", printk->printk) < 0)
   return NULL;
 
  return format;
-- 
1.7.1